user.go

```diff
@@ -1,33 +1,73 @@
 package buildah
 
 import (
+	"fmt"
 	"os/user"
 	"strconv"
+	"strings"
 
 	"github.com/opencontainers/runtime-spec/specs-go"
 )
 
-// TODO: we should doing these lookups using data that's actually in the container.
-func getUser(username string) (specs.User, error) {
-	if username == "" {
+func getUser(rootdir, username string) (specs.User, error) {
+	var gid64 uint64
+	var gerr error = user.UnknownGroupError("error looking up group")
+
+	userspec := strings.SplitN(username, ":", 2)
```

Comment:
How about 
user, group := strings.SplitN(username, ":", 2)
Then you don' t need to deal with the [0] stuff all over the place.